platforms/ide/ide/src/main/java/org/gradle/tooling/model/internal/gradle/IsolatedGradleProject.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling.model.internal.gradle;
+
+import org.gradle.tooling.model.BuildableElement;
+import org.gradle.tooling.model.ProjectIdentifier;
+import org.gradle.tooling.model.ProjectModel;
+import org.gradle.tooling.model.gradle.GradleScript;
+
+import java.io.File;
+
+/**
+ * Represents a Gradle project, isolated from the project hierarchy.
+ * <p>
+ * <b>This model is internal, and is NOT part of the public Tooling API.</b>
+ */
+public interface IsolatedGradleProject extends BuildableElement, ProjectModel {
```

Comment:
❔ Do we actually need to declare the interface for the model if it is entirely internal?

Reply:
We still need a model name to make a request via the intermediate model provider.
Should we then use just a binary name of the implementation class?

Reply:
I removed the interface, and use the implementation class name as the model name. This should be enough for internal purposes, and it also simplifies the intermediate model provider interface.